src/main/kotlin/com/learning/medicare/user/UserController.kt

```diff
@@ -45,17 +45,8 @@ class UserController(
     }
 
     @PostMapping("/{caregiverId}/patient")
-    fun associate(@PathVariable caregiverId: Long, @RequestBody payload: PatientDTO): ResponseEntity<TakesCareOf> {
-        return try {
-            ResponseEntity(userService.associate(payload.patient_id, caregiverId), HttpStatus.OK)
-        } catch (e: UserNotFoundException) {
-            ResponseEntity(null, HttpStatus.NOT_FOUND)
-        } catch (e: InvalidAssociationException) {
-            ResponseEntity(null, HttpStatus.BAD_REQUEST)
-        } catch (e: InvalidCaregiver) {
-            ResponseEntity(null, HttpStatus.BAD_REQUEST)
-        }
-    }
+    fun associate(@PathVariable caregiverId: Long, @RequestBody payload: PatientDTO) = userService.associate(payload.patient_id, caregiverId)
```

Comment:
I'm also confused on how to call the service/what to return from the controller. Even though I added a `@ControllerAdvice` to intercept exceptions, I don't it's getting called